lib/pegboard/manager/src/utils.rs

```diff
@@ -43,3 +83,167 @@ pub fn get_s3_path_stub(url: &Url, with_uuid: bool) -> Result<PathBuf> {
 
 	Ok(path_stub)
 }
+
+pub async fn init_sqlite_db(db_url: &str) -> Result<()> {
+	if !Sqlite::database_exists(db_url).await? {
+		Sqlite::create_database(db_url).await?;
+	}
+
+	Ok(())
+}
+
+pub async fn build_sqlite_pool(db_url: &str) -> Result<SqlitePool> {
+	SqlitePoolOptions::new()
+		.connect(db_url)
+		.await
+		.map_err(Into::into)
+}
+
+// TODO: Replace with migrations
+pub async fn init_sqlite_schema(pool: &SqlitePool) -> Result<()> {
+	// Attempt to use an existing connection
+	let mut conn = if let Some(conn) = pool.try_acquire() {
+		conn
+	} else {
+		// Create a new connection
+		pool.acquire().await?
+	};
+
+	sqlx::query(indoc!(
+		"
+		CREATE TABLE IF NOT EXISTS events (
+			data BLOB NOT NULL,
+			create_ts INTEGER NOT NULL
+		)
+		",
+	))
+	.execute(&mut *conn)
+	.await?;
+
+	// sqlx::query(indoc!(
+	// 	"
+	// 	CREATE TABLE IF NOT EXISTS commands (
+
+	// 	)
+	// 	",
+	// ))
+	// .execute(&mut *conn)
+	// .await?;
+
+	sqlx::query(indoc!(
+		"
+		CREATE TABLE IF NOT EXISTS containers (
+			container_id TEXT PRIMARY KEY, -- UUID
+			create_ts INTEGER NOT NULL,
+			stop_ts INTEGER,
+			exit_ts INTEGER,
+			exit_code INTEGER
+		)
+		",
+	))
+	.execute(&mut *conn)
+	.await?;
+
+	Ok(())
+}
+
+/// Executes queries and explicitly handles retry errors.
+pub async fn query<'a, F, Fut, T>(mut cb: F) -> Result<T>
+where
+	F: FnMut() -> Fut,
+	Fut: std::future::Future<Output = std::result::Result<T, sqlx::Error>> + 'a,
+	T: 'a,
+{
+	let mut i = 0;
+
+	loop {
+		match cb().await {
+			std::result::Result::Ok(x) => return Ok(x),
+			std::result::Result::Err(err) => {
+				use sqlx::Error::*;
+
+				if i > MAX_QUERY_RETRIES {
+					bail!("max sql retries: {err}");
```

Comment:
where do these errors get written & monitored? we need to be able to catch every tiny issue with pb monitors

Reply:
Was planning on having vector pick up the logs from the manager and stream them to the backend. Didn't get to it for MVP I don't think